nixos/modules/services/audio/navidrome.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.navidrome;
+in {
+  options = {
+    services.navidrome = {
+      enable = mkEnableOption pkgs.navidrome.meta.description;
+      settings = mkOption rec {
+        type = types.attrs;
+        apply = recursiveUpdate default;
+        default = {
+          Address = "127.0.0.1";
+          Port = 4533;
+        };
+        example = {
+          MusicFolder = "/mnt/music";
+        };
+        description = ''
+          Configuration for Navidrome, see <link xlink:href="https://www.navidrome.org/docs/usage/configuration-options/"/> for supported values
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.navidrome = {
```

Comment:
Wow looks like upstream has a great [unit](https://github.com/deluan/navidrome/blob/1be79fa945739c574151ddeb130588c94dff0dbb/contrib/navidrome.service) already worked out!

Maybe you can alter the `navidrome` package to include this unit and then use it instead of writing your own.

Bonus points awarded if you can file an issue upstream to have them fix their unit so it doesn't include a hard-coded reference to `/usr/bin`.

Reply:
+1 on reusing the upstream unit. The ExecStart containing an absolute path isn't a problem because we can simply override it.

Reply:
In my opinion using the upstream version does not bring a lot more as it's only defining default sanboxing systemd options

Reply:
The idea I'm working toward recently is to get people filing PRs against upstream unit files, which will improve the situation across all distros, decrease the maintenance burden in nixpkgs, add peer review from the author(s) of the software, and generally be a good thing.

Given that upstream actually ships a decent unit file it seems likely they would be willing to implement your changes if you filed a PR. This is the ideal scenario and we should really try to work **with** upstream instead of orthogonal to them. I kindly ask you at least reconsider your position on this please :bow:

Reply:
OK, I understand

Reply:
I opened a PR upstream with most of my changes here (except RootDirectory related setting as it is very distro dependent) https://github.com/deluan/navidrome/pull/677

Reply:
Fantastic! Thank you so much @nyanloutre :tada:

Reply:
Thank you @nyanloutre !